async_service/function_service/async_function_deployment.py

```diff
@@ -0,0 +1,159 @@
+import os
+from typing import Callable, Dict
+
+from fastapi import FastAPI, HTTPException
+
+from async_service.function_service.utils import (
+    INTERNAL_FUNCTION_NAME,
+    async_wrapper_func,
+    get_functions_dict_with_input_signatures,
+)
+from async_service.processor import Processor
+from async_service.types import (
+    InputMessage,
+    MessageProcessFailure,
+    OutputMessageTimeoutError,
+    WorkerConfig,
+)
+
+FUNCTION_SCHEMA_ENDPOINT = "/function-schemas"
+RESULT_ENDPOINT = "/result/{request_id}"
+
+
+class AsyncFunctionDeployment:
```

Comment:
Maybe call it `FunctionAsyncExecutor`? I don't have strong opinions here

This is not actually "Deployable" , and `AsyncFunction` can sometimes mean an actual async function (i.e. `async def `)

Reply:
Yeah, that will be better I guess